### DISCUSSION TITLE
No cycleway -> one-way only?

### DISCUSSION THREAD

**Discussion by AndreasHoffmann2** [streetcomplete/StreetComplete#3966] [General]
Hi,

I tried tagging a road with "no cycleway". The app did then ask me, if I am really sure that this one-way road can not be used in both directions with a bike.
What does this question have to do with the tag if there is a cycleway?
I don't get it.

![Screenshot_20220415_142250_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163574100-8c28b368-f697-4d4c-b460-f1fee9fc208d.jpg)
![Screenshot_20220415_142256_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163574107-d600e5e6-10f5-4ec6-94dd-5e1850a32a36.jpg)
![IMG_20220415_141849](https://user-images.githubusercontent.com/8135241/163574119-7ec69d18-9cbf-429c-8a7e-e0a61c748894.jpg)

**Comment by matkoniecz** [comment]
Look through selectable options. One of them is covering contraflow cycling (that you can cycle in both directions on oneway road)

**Reply by AndreasHoffmann2** [reply]
That option (the last one) means that there is some dedicated infrastructure for cyclists besides the lanes for cars, doesn't it?
But this street does not have any dedicated infrastructure.
![Screenshot_20220415_152506_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163576190-08209ef2-2f50-4af8-9eb5-049f8701b14e.jpg)

**Reply by matkoniecz** [reply]
Have you checked all options? There should be one for traffic in two directions, without own lane (select left side of the road)

**Reply by AndreasHoffmann2** [reply]
I think, I have:
![Screenshot_20220415_160946_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163588616-0859cefb-62ca-4686-b884-b04dafa35d84.jpg)
![Screenshot_20220415_160950_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163588625-467b8d63-15ca-4732-9124-d683c9d414f5.jpg)
![Screenshot_20220415_160932_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163588636-c5ca903f-faeb-4cf7-b9d4-cc7bcddb0d37.jpg)
![Screenshot_20220415_160937_de westnordost streetcomplete](https://user-images.githubusercontent.com/8135241/163588645-f20377e2-1b5d-442f-9f2a-0c365de241ff.jpg)

I don't really understand, why the choice "no cycling-infrastructure" should have an influence on the one-way-rules in a street. This seems like two independent things to me.

**Reply by matkoniecz** [reply]
"cyclists can use road in both directions" (contraflow) is a cheap[1] and very useful bicycle infrastructure 

[1]just few traffic signs are typically needed

**Reply by westnordost** [reply]
Well there it is

![image](https://user-images.githubusercontent.com/4661658/164333187-e3c82cf8-217e-4ac4-b544-e3fbc892cdf9.png)

**Reply by AndreasHoffmann2** [reply]
Thank you.
I see two issues with that.
First the image: why does the description differ from the image? Shouldn't there be a two-way-arrow? Even though it does only appear in the "wrong" direction.
That's probably the reason why I did not even find this answer while posting the screenshots (pretty embarrassing for me).
Perhaps the text could be changed instead of the icon: cyclists may drive in the opposite direction.

But even then: for me, the app's behavior is not transparent here.
On every normal road, I just look down to the ground and select whatever I see. And it does not have any influence on the allowed usage by bikes.

And suddenly, when I do the same thing on a one-way-road, selecting "none" changes that? Why?
The installed infrastructure and the signs at the intersection are two different things, aren't they?

This seems a lot like it should be split into two questions:
"Is there a cycleway?" 
"Are cyclists allowed to drive in the opposite direction here?"

The second question should only appear, if the first answer leaves "no" as a consistent answer to the second one.

**Reply by westnordost** [reply]
> And suddenly, when I do the same thing on a one-way-road, selecting "none" changes that? Why?
> The installed infrastructure and the signs at the intersection are two different things, aren't they?

Yes, they are two different things. Unfortunately, in OSM, these two different things are sometimes mixed together in one tag (`cycleway=opposite`).
**Also**, in some regions, the fact that the road may be used in contraflow by cyclists is denoted by pictograms that just look like that illustration.

**Reply by westnordost** [reply]
Would this help?

![image](https://user-images.githubusercontent.com/4661658/164338104-afacb5a6-9a94-4bbb-ac78-fadc82d52310.png)

**Reply by AndreasHoffmann2** [reply]
I'm not sure if this would have helped. While being in the street, it did not occur to me at all that there might be a special option for the "left side" of a one-way-road. Since there were no cars (downtown of Amsterdam), I did not even know the normal direction of travel there.

I have another thought that would definitely have helped me: change the popup-message.

"Are cyclists allowed to drive in the opposite direction in this one-way-road? In doubt, check if there is a sign at the end of the street." Yes/No/Abort

When selecting "yes" SC would silently upload the other option.

**Reply by westnordost** [reply]
What is the difference to the current message? How would it be better?

**Reply by AndreasHoffmann2** [reply]
When selecting "none" for both directions, I was in no way aware that I was limiting the usage by cyclists by that. I chose this option a gazillion times for normal roads. And it always meant that cyclists have to use the road then. Never did it mean that cyclists are not allowed to drive at all on this side.

Thus, when the message popped up, I had no idea, what I did wrong. Honestly, my first thought was some kind of a bug.

The new message is different: There are clear instructions what the user needs to do.

It's still not transparent, **why** the message pops up. But at least, the user can just answer it and move on.

The best option would IMO still be to decouple the two answers (as proposed above): Ask for the cycle-infrastructure first and for two-way-usage afterwards (if still necessary).
But you wrote that this is not compatible with OSM.

**Reply by westnordost** [reply]
Could you post this as a new ticket please? It is becoming a little long in this thread (that GitHub collapses threads after a certain length is no help)

**Reply by AndreasHoffmann2** [reply]
Did that: https://github.com/streetcomplete/StreetComplete/issues/3984
